rhobs/alerting/data_plane/prometheus.pipeline_alerts.yaml

```
@@ -146,3 +146,38 @@ spec:
             team: pipelines
             # link will be available once https://issues.redhat.com/browse/SRVKP-5899 is complete
             runbook_url: https://gitlab.cee.redhat.com/konflux/docs/sop/-/blob/main/pipeline-service/slos/chains-performance.md
+        - alert: PACControllerDeadlock
+          expr: |
+            sum by (source_cluster) (increase(pac_watcher_work_queue_depth[2m])) > 50
+          for: 75m
+          labels:
+            severity: critical
+            slo: "true"
+          annotations:
+            summary: >-
+              Tekton PAC controller appears to have stopped processing active pipelineruns.
+            description: >-
+              Tekton PAC controller on cluster {{ $labels.source_cluster }} has appeared deadlocked on {{ $value }} pipelinerun events.
+            alert_team_handle: <!subteam^S04PYECHCCU>
+            team: pipelines
+            # link will be available once https://gitlab.cee.redhat.com/konflux/docs/sop/-/merge_requests/163 merges
+            runbook_url: https://gitlab.cee.redhat.com/konflux/docs/sop/-/blob/main/pipeline-service/slos/tekton-pipeline-related-deadlocks.md
+        - alert: PACControllerPerformance
+          expr: |
+            (sum by (source_cluster) (increase(pac_watcher_client_latency_bucket{le='1'}[5m]))
```

```suggestion
            (sum by (source_cluster) (increase(pac_watcher_client_latency_bucket{le='10'}[5m]))
```

I have a PR up to change this for chains as well, so don't worry, just an adjustment on my part

---

So, do I need to do the changes? or will you be covering that in your chains PR?